### PR TITLE
Basic @next/font/google integration test

### PR DESCRIPTION
**Pull request by wbinnssmith**
This:

* Implements a basic http server to mock returning a stylesheet from the Google Fonts API (**Note**: This importantly does *not* mock returning the font resources themselves, which are currently loaded by the browser. We should implement something to do this when we implement caching arbitrary http resources traced from `url()` and this is no longer loaded by the test browser)
* Adds an integration test that asserts the basic shape of the JS object returned by font functions
* ~Adds an integration test that asserts the browser correctly loads a font for the ascii unicode range, along with the appropriate `font-display`, variant, etc.~ Unfortunately `document.fonts` is not reliable with different font-display loading patterns as any of them can fall back.

As noted, this does not mock responses from Google for the font resources themselves, which are currently loaded by the test browser. This means that we'll be dependent on this external service for passing integration tests until we implement caching of `url()`s in Turbopack. We should monitor the reliability of this test.

### CODE REVIEW

File: crates/next-dev/tests/integration.rs [hunk 1]
```diff
@@ -131,6 +132,19 @@ async fn run_test(resource: &str) -> JestRunResult {
...
     let requested_addr = get_free_local_addr().unwrap();
 
+    // Reference this here so it's dropped in this scope along with the browser and
+    // Turbopack server
+    let mut _font_server: Option<MockServer> = None;
+    let font_dir = path.join("__google_fonts__");
+    if font_dir.exists() {
+        let server = create_font_server(&font_dir);
+        env::set_var(
+            "TURBOPACK_TEST_ONLY_GOOGLE_FONTS_STYLESHEET_URL",
```
Comment (wbinnssmith) [inline crates/next-dev/tests/integration.rs:142]: I _think_ this should be okay with the way `cargo test`/`cargo nexttest` run tests in parallel, with multiple threads but only one test running in serial within the thread (see unsetting this value below)

File: crates/next-dev/tests/integration/next/font-google/basic/__google_fonts__/css2?family=Inter:ital,wght@0,100..900&display=optional [hunk 2]
```diff
@@ -0,0 +1,63 @@
+/* cyrillic-ext */
```
Comment (wbinnssmith) [inline crates/next-dev/tests/integration/next/font-google/basic/__google_fonts__/css2?family=Inter:ital,wght@0,100..900&display=optional:1]: Looks like this filename is invalid on Windows...

Comment (kwonoj) [reply crates/next-dev/tests/integration/next/font-google/basic/__google_fonts__/css2?family=Inter:ital,wght@0,100..900&display=optional:1]: Ugh yeah afaik : is not allowes char in Windows.

Comment (ForsakenHarmony) [reply crates/next-dev/tests/integration/next/font-google/basic/__google_fonts__/css2?family=Inter:ital,wght@0,100..900&display=optional:1]: I wouldn't put any special chars in the filename (except for `_` and `-`)

File: crates/next-dev/tests/integration.rs [hunk 3]
```diff
@@ -159,10 +162,15 @@ async fn run_test(resource: &str) -> JestRunResult {
         address = server.addr
     );
 
-    tokio::select! {
+    let result = tokio::select! {
         r = run_browser(server.addr) => r.expect("error while running browser"),
         _ = server.future => panic!("Never resolves"),
-    }
+        _ = mock_server_future => panic!("Never resolves"),
```
Comment (sokra) [inline crates/next-dev/tests/integration.rs:168]:
``` rs
        // Poll the mock_server first to add the env var
        _ = mock_server_future => panic!("Never resolves"),
        r = run_browser(server.addr) => r.expect("error while running browser"),
        _ = server.future => panic!("Never resolves"),
```

**Review by sokra — CHANGES_REQUESTED** [review]

**Review by sokra — APPROVED** [review]

**Comment by wbinnssmith** [comment]
@alexkirsz it looks like this test caught a regression from #3184, which made the next-font transform no longer run.

**Pull request merged**